programs/drift/src/instructions/lp_pool.rs

```diff
@@ -100,6 +105,105 @@ pub fn handle_update_constituent_target_weights<'c: 'info, 'info>(
     Ok(())
 }
 
+pub fn handle_update_lp_pool_aum<'c: 'info, 'info>(
+    ctx: Context<'_, '_, 'c, 'info, UpdateLPPoolAum<'info>>,
+) -> Result<()> {
+    let mut lp_pool = ctx.accounts.lp_pool.load_mut()?;
+    let state = &ctx.accounts.state;
+
+    let slot = Clock::get()?.slot;
+
+    let remaining_accounts = &mut ctx.remaining_accounts.iter().peekable();
+
+    let AccountMaps {
+        perp_market_map: _,
+        spot_market_map,
+        mut oracle_map,
+    } = load_maps(
+        remaining_accounts,
+        &MarketSet::new(),
+        &MarketSet::new(),
+        slot,
+        Some(state.oracle_guard_rails),
+    )?;
+
+    let constituent_map = ConstituentMap::load(&ConstituentSet::new(), remaining_accounts)?;
+
+    msg!("{}", constituent_map.0.len());
+    msg!("{}", lp_pool.constituents);
+
+    validate!(
+        constituent_map.0.len() == lp_pool.constituents as usize,
+        ErrorCode::WrongNumberOfConstituents,
+        "Constituent map length does not match lp pool constituent count"
+    )?;
+
+    let mut aum: u128 = 0;
+    for i in 0..lp_pool.constituents as usize {
+        let mut constituent = constituent_map.get_ref_mut(&(i as u16))?;
```

Comment:
this looks like it requires all of `lp_pool.constituents` to be in `remaining_accounts`, which doesn't match the logic in `driftClient.getUpdateDlpPoolAumIxs` which allows a subset of constituents